### PR TITLE
use the k8s-staging-images project to build cloud-provider-kind images

### PR DESCRIPTION
**Pull request by upodroid**
Part of https://github.com/kubernetes-sigs/cloud-provider-kind/pull/155

/cc @aojea @BenTheElder

### CODE REVIEW

File: config/jobs/image-pushing/k8s-staging-cloud-provider-kind.yaml [hunk 1]
```diff
@@ -14,8 +14,8 @@ postsubmits:
             command:
               - /run.sh
             args:
-              - --project=k8s-staging-kind
-              - --scratch-bucket=gs://k8s-staging-kind-gcb
+              - --project=k8s-staging-images
```
Comment (BenTheElder) [inline config/jobs/image-pushing/k8s-staging-cloud-provider-kind.yaml:17]: are we running _all_ cloudbuilds here? we're going to have quota issues fetching the logs etc and we're reducing the isolation of these subprojects??

Comment (BenTheElder) [reply config/jobs/image-pushing/k8s-staging-cloud-provider-kind.yaml:17]: (now someone's job that's only supposed to push some other subproject can also push to this staging location, right?)

lots of subprojects are not checking their logs closely when promoting images, if I understand what we're doing correctly: this is risky and we should go back to separate projects per subproject.

cc @kubernetes/sig-k8s-infra-leads

Comment (ameukam) [reply config/jobs/image-pushing/k8s-staging-cloud-provider-kind.yaml:17]: Are the IAM permissions/ACL defined [here](https://github.com/kubernetes/k8s.io/blob/main/infra/gcp/terraform/k8s-staging-images/iam.tf) not enough to handle this ?

Comment (upodroid) [reply config/jobs/image-pushing/k8s-staging-cloud-provider-kind.yaml:17]: The plan is for all subprojects to share the single cloudbuild project. The quotas are very high(something like 30 concurrent jobs). k/k will run cloudbuild jobs on its own project given its complexity. 

We enforce the expected behaviour by only allow promoting images from specified sources that we review. In the extremely unlikely situation that a maintainer misconfigures their cloudbuild definition and publishes it against another subprojects registry, the image can't be promoted and is deleted after 90d and they will spot the errors.

Also, humans only have privileges to push to their own respective registries and only prow and infra admins can submit cloudbuild jobs. The only way a cloudbuild job can be submitted by maintainers is via a postsubmit defined in the image pushing folder in the trusted cluster and commits to main branches approved by maintainers.

This approach is much simpler and with a reasonable set of guardrails.

**Pull request merged**